### PR TITLE
Added support for .tool-versions file in Gemfile for Ruby projects

### PR DESCRIPTION
**Pull request by kpumuk**
_This MR is related to https://github.com/dependabot/dependabot-core/issues/1033 in a way that it adds support for `.tool-versions` files for Ruby projects._

Starting with Bunder 2.4.20, it supports specifying Ruby version via `.tool-versions` (see https://github.com/rubygems/rubygems/pull/6898 and [Specifying a Ruby Version](https://bundler.io/guides/gemfile_ruby.html)). Unfortunately, Dependabot does not download this file and does not parse Ruby version from it.

This MR enables Ruby projects to specify Ruby version in a single file.

### CODE REVIEW

File: bundler/lib/dependabot/bundler/file_updater/lockfile_updater.rb
Comment (kpumuk) [inline bundler/lib/dependabot/bundler/file_updater/lockfile_updater.rb:282]: Perceived complexity check is triggered on this method.

One interesting thing about this is that this and previous lines are redundant, as the last line is `.reject(&:support_file?)` and both `.ruby-version` and `.tool-versions` are marked as support files.

File: bundler/spec/fixtures/projects/bundler1/tool_versions_file/.gitignore
Comment (kpumuk) [inline bundler/spec/fixtures/projects/bundler1/tool_versions_file/.gitignore:1]: `.tool-versions` is ignored at the root of the project, and so we need to negate it here to add this file to the fixture project.

File: bundler/lib/dependabot/bundler/file_updater/lockfile_updater.rb
Comment (kpumuk) [reply bundler/lib/dependabot/bundler/file_updater/lockfile_updater.rb:282]: Here is my proposal: https://github.com/kpumuk/dependabot-core/commit/3e370f1b8dadeab8048fb79de5beca2e3eaa3abf

Please let me know if this is the preferred approach, and I will merge it into this MR.

**Comment by janklimo** [comment]
Looking forward to this! Thank you @kpumuk

**Comment by kpumuk** [comment]
> Looking forward to this! Thank you @kpumuk

You and me both :-) I hope this will eventually get attention from maintainers.

**Comment by janklimo** [comment]
@deivid-rodriguez seeing you were [involved](https://github.com/rubygems/rubygems/pull/6898) in adding support for this in `rubygems`, can you help move this forward by reviewing or assigning the right person? Dependabot is still broken for us and many others:

Screenshot 2024-05-15 at 10 58 42

**Comment by deivid-rodriguez** [comment]
Hei, I no longer maintain Dependabot.

his seems tested and ready to be reviewed though, so since it's been hanging for a couple months, I think it's fine to politely ping some other maintainer and ask for a review.

The Dependabot team is usually super keen on taking community contributions, they are just very busy :sweat_smile:.

**Comment by janklimo** [comment]
@robaiken @landongrindheim you guys have done a lot of great work on the Ruby side of things of Dependabot. Would you have a moment to review/merge?

**Review by jurre — APPROVED** [review]

**Comment by kpumuk** [comment]
@jurre rebased main and reworded test to address lint error in https://github.com/dependabot/dependabot-core/actions/runs/9117390435/job/25068129813

**Comment by sachin-sandhu** [comment]
Hi @jurre , Request you to have a quick review and approve as @kpumuk has added new commits after last approval. Thank you.

**Comment by jurre** [comment]
Thanks everyone!

**Comment by bewuethr** [comment]
Omg it's been merged! Thank you all 🥳

**Pull request merged**
